docs/basics/101-114-filesystem.rst

```diff
@@ -0,0 +1,127 @@
+
+   $ datalad status
+
+We can see that the old file is marked as ``deleted``, and
+simultaneously, an ``untracked`` file appears: the renamed
+PDF.
+
+In theory, we would need to do a ``datalad save`` at this point
+to commit the "deletion" and the "new file". This however, is
+suboptimal, and Git has built-in commands that provide a better
+solution.
+
+Lets revert the renaming of the the files:
+
+.. runrecord:: _examples/DL-101-114-103
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git reset --hard master       # this restores TLCL.pdf
+   $ rm The_Linux_Command_Line.pdf       # this removes the untracked copy
+
+A more convenient way to rename files is the ``git mv`` command:
+
+.. runrecord:: _examples/DL-101-114-104
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git mv TLCL.pdf The_Linux_Command_Line.pdf
+
+.. runrecord:: _examples/DL-101-114-105
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ datalad status
+
+We can see that the old file is still seen as "deleted", but the "new",
+renamed file is "added". A ``git status`` displays the change
+in the dataset a bit more accurate:
+
+.. runrecord:: _examples/DL-101-114-106
+   :language: console
+   :workdir: dl-101/DataLad-101/books
+
+   $ git status
+
+.. todo::
+
+   can I mitigate a ``git commit`` here? I believe the best ``datalad save``
+   command to save the renaming (without the rest of the dataset state) is
+   ``datalad save -m "renamed TLCL to verbose title" The_Linux_Command_Line.pdf TLCL.pdf``,
+   but this still requires listing all paths, and tab completion does not seem to work for
+   the "deleted" files...
+
+TODO
+
+What happens if I rename a directory or subdataset?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What happens if I move files, directories, or subdatasets?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The first question is easy. Because in Unix renaming is the same as moving,
+moving works exactly like renaming in the examples with the PDFs described above.
+Make sure to use ``git mv`` instead of ``mv``.
+
+TODO: directory (all files in a directory will be renamed)
+TODO: subdataset (tricky: change in .gitmodules. Also, I failed trying to
+revert a ``git mv`` with ``git reset --hard master``. It did not move
+subdataset back into original place, the subds became an untracked directory.
+
+What if I move the superdataset into a different place on my computer?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TODO: I don't think it should be a problem, but not sure whether I'm
+missing something
+
+What happens if I copy files?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TODO: Copying will create a new file that could be added. Its not symlinked,
+I don't yet understand why, but the file is write-protected (why exactly? bc
+symlinked content is write protected?), which is
+inconveniently difficult to explain (needs digression into permissions)
```

Comment:
Continuing the example from above:

```
$ cp two copyoftwo
$ datalad save
```

should work just fine. On save, git-annex will automatically identify any identical files and symlink them to the same annex object.